### PR TITLE
174945873 — Fix The :finder_sql association option is deprecated

### PR DESCRIPTION
**Pull request by knowuh**
* removed `:finder_sql` from investigations and activities
* removed `:finder_sql` from pages
* removed `:finder_sql` from sections

Most of the `finder_sql` was from some meta-programming that dynamically created AR associations for embeddables.  I pulled these common relations up into a `has_embeddable` mixin. 

Some older hand-crafted proc scopes generated SQL that is identical to clean chained scoped relations.

I added @scytacki to this PR because it touched many models.

[#174945873]
https://www.pivotaltracker.com/story/show/174945873

### CODE REVIEW

File: rails/app/models/activity.rb
Comment (dougmartin) [inline rails/app/models/activity.rb:27]: Is this new method still needed?

**Review by dougmartin — APPROVED** [review]
Looks good other than wondering if the new `student_only` method added to `pages` is needed.  I'll let @scytacki have the final review say.

File: rails/app/models/activity.rb
Comment (knowuh) [reply rails/app/models/activity.rb:27]: Thanks -- no, this must have been one of the experiments we were working on that I fogot to back out.  I will remove it.

Comment (knowuh) [reply rails/app/models/activity.rb]: edad8bdaf2281c62f044ea53ba75db802e406537 undoes this.

**Comment by knowuh** [comment]
@scytacki we thought you should have a look at the PR when you get a chance.  It removes some of the class eval magic from including embeddables, and updates the student scopes to be more rails4-llike.

**Comment by dougmartin** [comment]
@knowuh / @scytacki I'm going to go ahead and merge this so we can clean up the test output.

**Pull request merged**
